File: src/test/testData/java/implicit/HashSet.java
package ru.ustinov;

import java.net.URL;
import java.util.HashSet;

class Simple {
    public static void main() {
        <warning descr="This data structure can probably implicitly invoke URL#hashCode()">HashSet<URL> set;</warning>
        HashSet<Integer> set2;
        <warning descr="This data structure can probably implicitly invoke URL#hashCode()">HashSet<HashSet<URL>> set3, set4;</warning>
        HashSet<HashSet<Integer>> set5, set6;
    }
}

File: src/test/testData/kotlin/implicit/HashSet.kt
import java.net.URL
import java.util.HashMap
import java.util.HashSet

@Suppress("UNUSED_VARIABLE")
fun main() {
    <warning descr="This data structure can probably implicitly invoke URL#hashCode()">val a: HashMap<URL, Int></warning>
    val b: HashMap<Int, URL>
    <warning descr="This data structure can probably implicitly invoke URL#hashCode()">val c: HashMap<HashSet<URL>, Int></warning>
    val d: HashMap<Int, HashSet<Int>>
}